packages/core/src/modules/user/auth0-synchronization-service.ts

```diff
@@ -0,0 +1,104 @@
+import { type Logger, getLogger } from "@dotkomonline/logger"
+import type { User, UserWrite } from "@dotkomonline/types"
+import { addDays } from "date-fns"
+import { Auth0UserNotFoundError } from "../external/auth0-errors"
+import type { Auth0Repository } from "../external/auth0-repository"
+import type { UserService } from "./user-service"
+
+export interface Auth0SynchronizationService {
+  update(payload: UserWrite): Promise<User>
+  handleUserSync(sub: string, now: Date): Promise<User>
+  handlePopulateUserWithFakeData(auth0Id: string, email?: string | null): Promise<void>
```

Comment:
Small nit, but I don't think prefixing actions with `handle` adds any meaning to the function name. Perhaps just do `synchronizeUser()` and `populateUserWithFakeData()`.

Can you also add a doc comment on `populateUserWithFakeData()` on why exactly we have a method involving fake data? (Just so others that don't know the ins and out of the auth0 onboarding can understand?)